src/tools/linkchecker/main.rs

```diff
@@ -110,10 +113,22 @@ macro_rules! t {
     };
 }
 
+#[derive(Parser)]
+struct Cli {
+    docs: PathBuf,
+    #[clap(long)]
+    link_targets_dir: Vec<PathBuf>,
+}
+
 fn main() {
-    let docs = env::args_os().nth(1).expect("doc path should be first argument");
-    let docs = env::current_dir().unwrap().join(docs);
-    let mut checker = Checker { root: docs.clone(), cache: HashMap::new() };
+    let mut cli = Cli::parse();
+    cli.docs = cli.docs.canonicalize().unwrap();
```

Comment:
To avoid the dependencies, would you be willing to instead do something like this?

```rust
let mut link_targets_dir = Vec::new();
let mut docs_path = None;
for arg in std::env::args().skip(1) {
    if let Some(path) = arg.strip_prefix("--link-targets-dir=") {
        link_targets_dir.push(PathBuf::from(path));
    } else {
        if docs_path.replace(PathBuf::from(arg)).is_some() {
            panic!("docs path should be specified once");
        }
    }
}
let docs_path =
    std::path::absolute(docs_path.expect("docs path should be the last argument")).unwrap();
```

Reply:
Yeah, I can integrate something like that in a bit.